File: functions/liboqs/variants/go/0_fpr_half_9.go
package main

func fpr_half(x uint64) uint64 {
    if (x>>52) & 0x7FF == 0 {
        return x >> 1
    }
    return (x & ((1 << 52) - 1)) | (((x >> 52) - 1) & 0x7FF) << 52
}


func main(){
    fpr_half(0)
}

File: functions/liboqs/variants/c/0_fpr_half_9.c
#include <stdint.h>

static uint64_t fpr_half(uint64_t x) {
    uint64_t mask = x == 0 ? 0 : 0xFFFFFFFFFFFFFFFF;
    x -= mask & ((uint64_t)1 << 52);
    mask = (((uint32_t)(x >> 52) & 0x7FF) + 1) >> 11;
    return x & (uint64_t)mask - 1;
}


int main(){
    fpr_half(0);
}
